packages/router/src/bindings/contractReader/index.ts

```diff
@@ -109,6 +151,9 @@ export const handleActiveTransactions = async (
       transaction.crosschainTx.invariant.transactionId,
     );
 
+    // if transfer is receiver prepared, just no router action required
+    if (transaction.status === CrosschainTransactionStatus.ReceiverPrepared) continue;
+
```

Comment:
probably a bit more logically consistent to add this status-based check to the `handleSingle` function